nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                                    Ok(config) => config,
+                                    Err(e) => {
+                                        error!(
+                                            log,
+                                            "error while fetching bgp peer config from db";
+                                            "location" => %location,
+                                            "port_name" => %port.port_name,
+                                            "error" => %e,
+                                        );
+                                        continue;
+                                    },
+                                };
+                                vacant_entry.insert((bgp_config_id, config.clone()));
+                                config
+                            },
+                        };
+
+                        //
+                        // build a list of prefixes from the announcements in the bgp config
+                        //
+
+                        // Same thing as above, check to see if we've already built the announce set,
+                        // if so we'll skip this step
+                        if bgp_announce_prefixes.get(&bgp_config.bgp_announce_set_id).is_none() {
+                            let announcements = match self
+                                .datastore
+                                .bgp_announce_list(
+                                    opctx,
+                                    &params::BgpAnnounceSetSelector {
+                                        name_or_id: bgp_config
+                                            .bgp_announce_set_id
+                                            .into(),
+                                    },
+                                )
+                                .await
+                            {
+                                Ok(a) => a,
+                                Err(e) => {
+                                    error!(
+                                        log,
+                                        "error while fetching bgp announcements from db";
+                                        "location" => %location,
+                                        "bgp_announce_set_id" => %bgp_config.bgp_announce_set_id,
+                                        "error" => %e,
+                                    );
+                                    continue;
+                                },
+                            };
+
+                            let mut prefixes: Vec<Prefix4> = vec![];
+
+                            for announcement in &announcements {
+                                let value = match announcement.network.ip() {
+                                    IpAddr::V4(value) => value,
+                                    IpAddr::V6(a) => {
+                                        error!(log, "bad request, only ipv4 supported at this time"; "requested_address" => ?a);
```

Comment:
Just a note that we should validate this in the API call handler and never actually get here. We should file an issue for this if there is not one already.

Reply:
Related:
* https://github.com/oxidecomputer/omicron/issues/5085
* https://github.com/oxidecomputer/omicron/pull/5107

Reply:
I'll check to see if there is an issue specific to BGP as well.